### PR TITLE
fix #1228, add java interface for Graph API

### LINKED ISSUE #1228: Stablize the java interface and scala interface
1. For java interface, it is better that user written streaming app don't need to be aware of scala or akka version. 
2. It should provide simpler and more intuitive way to write streaming app in java.

### PR DESCRIPTION
**Pull request by clockfly**

### CODE REVIEW

**Comment by manuzhang** [comment]
Could we have a simple example using java api ?

**Comment by clockfly** [comment]
The example will be hosted at https://github.com/gearpump/gearpump-java-example

**Comment by manuzhang** [comment]
I think a simple in-gearpump java example that evolves with gearpump api and is always up to date would be valuable. Besides, it would be a good guide for whoever wants to contribute to https://github.com/gearpump/gearpump-java-example

**Comment by clockfly** [comment]
Good suggestion.

Patch updated.

File: examples/streaming/wordcount-java/src/main/java/io/gearpump/streaming/examples/wordcountjava/WordCount.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    // For split task, we config to create two tasks
+    int splitTaskNumber = 2;
+    Processor split = new Processor(Split.class).withParallelism(splitTaskNumber);
+
+    // For sum task, we have two summer.
+    int sumTaskNumber = 2;
+    Processor sum = new Processor(Sum.class).withParallelism(sumTaskNumber);
+
+    UserConfig conf = UserConfig.empty();
+    StreamApplication app = new StreamApplication("wordcountJava", conf, system);
```
Comment (huafengw) [inline examples/streaming/wordcount-java/src/main/java/io/gearpump/streaming/examples/wordcountjava/WordCount.java]: Why user have to construct an ActorSystem to build a StreamApplication?

Comment (clockfly) [inline examples/streaming/wordcount-java/src/main/java/io/gearpump/streaming/examples/wordcountjava/WordCount.java]: Good point

**Comment by clockfly** [comment]
@huafengw 

Please check the new API design.

File: examples/streaming/wordcount-java/src/main/java/io/gearpump/streaming/examples/wordcountjava/Split.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    this.userConf = userConf;
+  }
+
+  private Long now() {
+    return System.currentTimeMillis();
+  }
+
+  @Override
+  public void onStart(StartTime startTime) {
+    self().tell(new Message("start", now()), self());
```
Comment (manuzhang) [inline examples/streaming/wordcount-java/src/main/java/io/gearpump/streaming/examples/wordcountjava/Split.java]: maybe wrap this in a method of super class like

```java
   public void outputToSelf(Message msg) {
      self().tell(msg, self());
   }
```

Comment (clockfly) [inline examples/streaming/wordcount-java/src/main/java/io/gearpump/streaming/examples/wordcountjava/Split.java]: I don't think it is cleaner. I think we should align with how akka does this.

File: examples/streaming/wordcount-java/src/main/java/io/gearpump/streaming/examples/wordcountjava/WordCount.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.gearpump.streaming.examples.wordcountjava;
+
+import akka.actor.ActorSystem;
```
Comment (manuzhang) [inline examples/streaming/wordcount-java/src/main/java/io/gearpump/streaming/examples/wordcountjava/WordCount.java]: please clean up the imports

**Comment by huafengw** [comment]
Kind of inconsistent that we have ```ClientContext.submit(app)``` in scala while ```app.submit(clientContext)``` in java. But generally I'll +1 with this PR.

**Comment by clockfly** [comment]
@huafengw

Please check the new look and feel. Now it is consistent with scala.

**Comment by huafengw** [comment]
+1

**Pull request merged**